File: horizontal_v1/codex/frontend/src/service/api/rbac.ts
import http from '../http'
import type { PageResult } from '../types'

export interface PageQuery {
  page?: number
  size?: number
  keyWord?: string
}

export interface UserPageQuery extends PageQuery {
  status?: number
  departmentIds?: number[]
}

export interface UserItem {
  id: number
  username: string
  name?: string | null
  nickName?: string | null
  phone?: string | null
  email?: string | null
  remark?: string | null
  status: number
  departmentId?: number | null
  departmentName?: string | null
  roleName?: string | null
}

export interface RoleItem {
  id: number
  name: string
  label?: string | null
  remark?: string | null
  relevance: boolean
}

export interface DeptItem {
  id: number
  name: string
  parentId?: number | null
  orderNum: number
  parentName?: string | null
}

export interface DeptTreeNode {
  id: number
  name: string
  parentId?: number | null
  orderNum: number
  children: DeptTreeNode[]
}

export const fetchUsers = (payload: UserPageQuery = { page: 1, size: 20 }) =>
  http.post<PageResult<UserItem>>('/rbac/users/page', {
    page: 1,
    size: 20,
    ...payload,
  })

export const fetchRoles = (payload: PageQuery = { page: 1, size: 20 }) =>
  http.post<PageResult<RoleItem>>('/rbac/roles/page', {
    page: 1,
    size: 20,
    ...payload,
  })

export const fetchDepts = () => http.post<DeptItem[]>('/rbac/depts/list', {})

export const fetchDeptTree = () => http.post<DeptTreeNode[]>('/rbac/depts/tree', {})

export const fetchPerms = () => http.get<string[]>('/rbac/perms')

File: horizontal_v1/codex/backend/app/schemas/rbac.py
from __future__ import annotations

from pydantic import Field

from backend.app.schemas.base import Schema


class EmptyRequest(Schema):
    pass


class DeptCreateRequest(Schema):
    name: str
    parent_id: int | None = None
    order_num: int = 0


class DeptOut(Schema):
    id: int
    name: str
    parent_id: int | None = None
    order_num: int = 0
    parent_name: str | None = None


class DeptTreeNode(Schema):
    id: int
    name: str
    parent_id: int | None = None
    order_num: int = 0
    children: list["DeptTreeNode"] = Field(default_factory=list)


DeptTreeNode.model_rebuild()


class DeptOrderItem(Schema):
    id: int
    parent_id: int | None = None
    order_num: int = 0


class PageRequest(Schema):
    page: int = 1
    size: int = 20
    key_word: str | None = None


class Pagination(Schema):
    total: int
    page: int
    size: int


class PageResult(Schema):
    list: list
    pagination: Pagination


class RoleCreateRequest(Schema):
    name: str
    label: str
    remark: str | None = None
    relevance: bool = False


class RoleOut(Schema):
    id: int
    name: str
    label: str | None = None
    remark: str | None = None
    relevance: bool = False


class UserCreateRequest(Schema):
    username: str
    password: str
    name: str | None = None
    nick_name: str | None = None
    head_img: str | None = None
    phone: str | None = None
    email: str | None = None
    remark: str | None = None
    status: int = 1
    department_id: int | None = None
    role_id_list: list[int] = Field(default_factory=list)


class UserUpdateRequest(Schema):
    id: int
    password: str | None = None
    name: str | None = None
    nick_name: str | None = None
    head_img: str | None = None
    phone: str | None = None
    email: str | None = None
    remark: str | None = None
    status: int | None = None
    department_id: int | None = None
    role_id_list: list[int] | None = None


class UserDisableRequest(Schema):
    id: int
    status: int


class UserMoveRequest(Schema):
    department_id: int
    user_ids: list[int]


class UserPageRequest(Schema):
    page: int = 1
    size: int = 20
    key_word: str | None = None
    status: int | None = None
    department_ids: list[int] = Field(default_factory=list)


class UserOut(Schema):
    id: int
    username: str
    name: str | None = None
    nick_name: str | None = None
    head_img: str | None = None
    phone: str | None = None
    email: str | None = None
    remark: str | None = None
    status: int
    department_id: int | None = None
    department_name: str | None = None
    role_name: str | None = None
